forge-game/src/main/java/forge/game/spellability/AbilityManaPart.java

```diff
@@ -422,10 +396,6 @@ public boolean meetsManaRestrictions(final SpellAbility sa, String restrictions)
                 }
             }
 
-            if (restriction.equals("CantCastNonArtifactSpells")) {
-                return !sa.isSpell() || sa.getHostCard().isArtifact();
-            }
-
             // the payment is for a resolving SA, currently no other restrictions would allow that
             if (getSourceCard().getGame().getStack().getInstanceMatchingSpellAbilityID(sa.getRootAbility()) != null) {
```

Comment:
@tool4ever can you check for what issue this check was for?

because i removed some restrictions above to use the lower isValid one?

Reply:
hmn this refactor certainly breaks the logic
e. g. you can no longer pay for _Convolute_ UnlessCost with _Thran Turbine_

Reply:
ah UnlessCost was the bad thing i forgot, i tested all the other ones

can't we just remove the InstanceMatching check?

Reply:
Unfortunately I added it on the old Gitlab
you can try and I'll see if I can break it without

Reply:
i think this might need bigger rewrite ...
currently we don't differ correct between using mana for Cost (Spell/Ability) and using mana for Effect at this point

we probably need to pass the "forEffect" Parameter down to this function

Reply:
especially with "Tin Street Gossip" saying: 
> Spend this Mana only to cast face-down spells or to turn creatures face up

we might need to wait for CR or oracle updates

Reply:
i might revert this change for now